app/cards/renaissance/border_guard.js

```diff
@@ -0,0 +1,160 @@
+BorderGuard = class BorderGuard extends Card {
+
+    types() {
+        return ['action']
+    }
+
+    coin_cost() {
+        return 2
+    }
+
+    play(game, player_cards) {
+        game.turn.actions += 1
+        let lantern_index = _.findIndex(player_cards.artifacts, function (artifact) {
+            return artifact.name === 'Lantern'
+        })
+
+        let cards_to_reveal = lantern_index > -1 ? 3 : 2;
+
+        if (_.size(player_cards.deck) === 0 && _.size(player_cards.discard) === 0) {
+            game.log.push(`&nbsp;&nbsp;<strong>${player_cards.username}</strong> has no cards in deck`)
+        } else {
+            player_cards.revealed = _.take(player_cards.deck, cards_to_reveal)
+            player_cards.deck = _.drop(player_cards.deck, cards_to_reveal)
+
+            let revealed_card_count = _.size(player_cards.revealed)
+            if (revealed_card_count < cards_to_reveal && _.size(player_cards.discard) > 0) {
+                DeckShuffler.shuffle(game, player_cards)
+                player_cards.revealed = player_cards.revealed.concat(_.take(player_cards.deck, cards_to_reveal - revealed_card_count))
+                player_cards.deck = _.drop(player_cards.deck, cards_to_reveal - revealed_card_count)
+            }
+
+            let are_all_actions = (_.size(player_cards.revealed) === cards_to_reveal) && _.every(player_cards.revealed, function (revealed_card) {
+                return _.includes(_.words(revealed_card.types), 'action')
+            });
+
+            let turn_event_id = TurnEventModel.insert({
```

Comment:
I think it would be helpful to wrap this chunk in a conditional to help speed things up and improve logging feedback.

Something like:

```
if (_.size(player_cards.revealed) > 1) {
    let turn_event_id = TurnEventModel.insert({
        game_id: game._id,
        player_id: player_cards.player_id,
        username: player_cards.username,
        type: 'choose_cards',
        player_cards: true,
        instructions: 'Choose a card to put in hand:',
        cards: player_cards.revealed,
        minimum: 1,
        maximum: 1
    })
    let turn_event_processor = new TurnEventProcessor(game, player_cards, turn_event_id)
    turn_event_processor.process(BorderGuard.put_in_hand)
} else if (_.size(player_cards.revealed) === 1) {
    BorderGuard.put_in_hand(game, player_cards, player_cards.revealed)
} else {
    game.log.push(`&nbsp;&nbsp;but <strong>${player_cards.username}</strong> has no cards in their deck`)
}
```